### PR TITLE
`Development`: Move hazelcast access to `@PostConstruct` to prevent potential slow startup

### PR DESCRIPTION
**Pull request by krusche**
Make sure hazelcast data (with potential network delay) is only accessed after all beans (services, resources) are initialized. Therefore, move all access to map, topic, etc. into methods with `@PostConstruct` annotations

#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

## Summary by CodeRabbit

- **Refactor**
	- Streamlined service initialization and configuration by integrating Hazelcast for distributed data management.
- **New Features**
	- Enhanced system reliability and performance by optimizing data setup and listener settings.
- **Tests**
	- Revised tests to accommodate the new initialization approaches and Hazelcast integration.
- **Documentation**
	- Provided additional explanations on direct access and initialization practices within the Hazelcast framework.

### CODE REVIEW

**Review by mateusmm01 — APPROVED** [review]
Code changes look good to me! Also tested on staging and ts6 and works as expected

File: src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/CacheHandler.java [hunk 1]
```diff
@@ -20,6 +20,8 @@ public abstract class CacheHandler<K, C extends Cache> {
     protected final IMap<K, C> cache;
 
     protected CacheHandler(HazelcastInstance hazelcastInstance, String name) {
+        // NOTE: it's ok to directly access hazelcastInstance in the constructor, because this is no Bean/Service and will be created after the services are initialized in
```
Comment (rstief) [inline src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/CacheHandler.java:23]: services are initialized in ... ?

**Review by rstief — COMMENTED** [review]
looks good, just a comment seems incomplete :)

File: src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/CacheHandler.java (hunk 1 above)
Comment (laurenzfb) [reply src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/CacheHandler.java:23]: i think the "in" is unnecessary, will remove it

**Review by laurenzfb — APPROVED** [review]
Tested locally, works as expected. Code also lgtm

**Review by mateusmm01 — APPROVED** [review]

**Review by rstief — APPROVED** [review]
Code looks good, thanks for fixing

**Review by nityanandaz — APPROVED** [review]
Code review: looks good

**Review by reschandreas — APPROVED** [review]
tested locally, works as expected. code also lgtm

**Review by undernagruzez — APPROVED** [review]
Tested locally, works as expected

**Pull request merged**